### PR TITLE
D2IQ-87203: react-popper console errors

### PR DESCRIPTION
**Pull request by mstruebing**
## Testing

You would need to build rebuild `ui-kit` and manually copy the `dist` folder to `kommander`, I do it like this from `ui-kit`:

```bash
npm run dist && \
 rm -rf ~/projects/work/kommander-ui/kommander/frontend/node_modules/@dcos/ui-kit/dist/packages && \
 cp -r dist/packages ~/projects/work/kommander-ui/kommander/frontend/node_modules/@dcos/ui-kit/dist/
```

Make sure to update the paths to yours.

http://localhost:1234/workspace/kommander-workspace/access-control/roles/new and select a resource type.

## Trade-offs

I can not see any difference without the `getFlipModifier` applied, anybody knows in which situation we need it?

## Checklist for PR creator

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [ ] If this PR is associated with a JIRA, it is mentioned in commit message footer ("Closes …")
- [ ] If this PR contains breaking changes, is stated in commit message body ("BREAKING CHANGE: …")
- [ ] Info for applicable sections above is provided

### CODE REVIEW

**Comment by seialkali** [comment]
I've done some digging and I've worked out what the `getFlipModifier` function is for. [`Flip` is a built-in Popper.js modifier](https://popper.js.org/docs/v2/modifiers/flip/) which `can change the placement of a popper when it's scheduled to overflow a given boundary`. Popper.js also allows you to add custom modifiers or to override the options of a built-in modifier. What `getFlipModifier` is doing is overriding the default options for the Flip modifier, to allow for the directions passed in via the `preferredDirections` prop to be used as the `fallbackPlacements` instead. This basically means that if you set a `preferredDirections` prop, it's not going to work for the situations where Flip comes into effect.

**Comment by vacas5** [comment]
Is there an opportunity for us to remove a dependency here? 

Secondly

File: packages/dropdownable/components/Dropdownable.tsx [hunk 1]
```diff
@@ -90,7 +75,7 @@ const Dropdownable: React.FC<DropdownableProps> = ({
     React.useState<HTMLDivElement | null>(null);
   const { styles, attributes } = usePopper(referenceElement, popperElement, {
     placement: getPreferredDirection(preferredDirections),
-    modifiers: [maxSize, applyMaxSize, getFlipModifier(preferredDirections)]
```
Comment (vacas5) [inline packages/dropdownable/components/Dropdownable.tsx:93]: I think this is just deprecating functionality which means it would be a BREAKING CHANGE.

**Comment by nataliepina** [comment]
Much of the changes related to what is being removed can be traced back to here - https://github.com/dcos-labs/ui-kit/pull/542/files

File: packages/dropdownable/components/Dropdownable.tsx (hunk 1 above)
Comment (mstruebing) [reply packages/dropdownable/components/Dropdownable.tsx:93]: I fixed it in another way now.

**Review by nataliepina — APPROVED** [review]

File: packages/dropdownable/components/Dropdownable.tsx [hunk 2]
```diff
@@ -90,7 +90,12 @@ const Dropdownable: React.FC<DropdownableProps> = ({
...
+    modifiers: [
+      maxSize,
+      applyMaxSize,
+      getFlipModifier(preferredDirections)
+      // we need valid modifiers here otherwise we get a bunch of console errors
+    ].filter(Boolean) as Array<Partial<Modifier<string, object>>>
```
Comment (seialkali) [inline packages/dropdownable/components/Dropdownable.tsx:98]: What's this filter for?

**Review by seialkali — APPROVED** [review]
Nice work figuring this one out! 🎉

**Pull request merged**